packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -138,13 +196,37 @@ export class Fleet extends Resource implements IFleet {
       throw new Error('baseCapacity must be greater than or equal to 1');
     }
 
+    if (props.computeConfiguration) {
+      if (props.computeType !== FleetComputeType.ATTRIBUTE_BASED_COMPUTE) {
+        throw new Error(`'computeConfiguration' can only be specified if 'computeType' is 'ATTRIBUTE_BASED_COMPUTE', got: ${props.computeType}`);
+      }
+      const diskGiB = props.computeConfiguration.disk?.toGibibytes();
+      if (diskGiB !== undefined && (diskGiB < 1 || !Number.isInteger(diskGiB))) {
+        throw new Error(`disk size must be greater than 1 GiB and an integer, got: ${diskGiB} GiB`);
+      }
+      const memoryGiB = props.computeConfiguration.memory?.toGibibytes();
+      if (memoryGiB !== undefined && (memoryGiB < 1 || !Number.isInteger(memoryGiB))) {
+        throw new Error(`memory size must be greater than 1 GiB and an integer, got: ${memoryGiB} GiB`);
+      }
+      const vCpu = props.computeConfiguration.vCpu;
+      if (vCpu !== undefined && !Token.isUnresolved(vCpu) && !Number.isInteger(vCpu)) {
+        throw new Error(`vCPU count must be an integer, got: ${vCpu}`);
+      }
+    }
+
     super(scope, id, { physicalName: props.fleetName });
 
     const resource = new CfnFleet(this, 'Resource', {
       name: props.fleetName,
       baseCapacity: props.baseCapacity,
       computeType: props.computeType,
       environmentType: props.environmentType,
+      computeConfiguration: props.computeConfiguration ? {
+        disk: props.computeConfiguration.disk?.toGibibytes(),
+        machineType: props.computeConfiguration.machineType,
+        memory: props.computeConfiguration.memory?.toGibibytes(),
+        vCpu: props.computeConfiguration.vCpu,
```

Comment:
Despite what the CloudFormation documentation says, the numeric props are not optional, see https://github.com/nmussy/aws-cdk/blob/6120d8f54103f81c4ec601221c1f56e5a4549236/packages/%40aws-cdk-testing/framework-integ/test/aws-codebuild/test/integ.project-fleet-attribute-based-compute.ts#L67-L73

Could you give them a default value, like [this for example](https://github.com/aws/aws-cdk/pull/32574/files/6120d8f54103f81c4ec601221c1f56e5a4549236#diff-5425be2f47cbad47acd51ad8eaa07758e9765df8bb28f242cade9f85f6c3d665R222-R224), and add an integ with just `{ machineType: GENERAL }`?